tools/lambda-promtail/lambda-promtail/sns.go

```diff
@@ -0,0 +1,21 @@
+package main
+
+import (
+	"context"
+
+	"github.com/aws/aws-lambda-go/events"
+)
+
+func processSNSEvent(ctx context.Context, evt *events.SNSEvent) error {
+	for _, record := range evt.Records {
+		event, err := stringToRawEvent(record.SNS.Message)
+		if err != nil {
+			return err
+		}
+		err = handler(ctx, event)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
I think we could move this and the sqs code to the `s3.go` file. 

Can you also include a small test for the sns event parsing.

Reply:
Yep, I can get those moved into `s3.go`.

I'm not 100% sure how to test since the logic is all in `handler` and `stringToRawEvent`. `stringToRawEvent` has a test but `handler` doesn't. I can try to come up with something but if you have any suggestions I'd appreciate it!

Reply:
@cstyan I had to do a little bit of rework but I got tests for both the SNS and SQS processing functions done. I also fixed a panic in the S3 parsing that I ran into while testing.